pkg/sql/tests/sysbench_test.go

```diff
@@ -0,0 +1,624 @@
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/cockroachdb/errors"
+	"github.com/jackc/pgx/v5"
+)
+
+const (
+	sysbenchTables          = 10    // how many tables in the dataset
+	sysbenchRowsPerTable    = 10000 // how many rows per table in the dataset
+	sysbenchRangeSize       = 100   // how many rows to scan during range scans
+	sysbenchPointSelects    = 10    // how many point selects to perform per transaction
+	sysbenchSimpleRanges    = 1     // how many simple range scans to perform per transaction
+	sysbenchSumRanges       = 1     // how many sum range scans to perform per transaction
+	sysbenchOrderRanges     = 1     // how many order range scans to perform per transaction
+	sysbenchDistinctRanges  = 1     // how many distinct range scans to perform per transaction
+	sysbenchIndexUpdates    = 1     // how many index updates to perform per transaction
+	sysbenchNonIndexUpdates = 1     // how many non-index updates to perform per transaction
+	sysbenchDeleteInserts   = 1     // how many delete-inserts to perform per transaction
+)
+
+// tableNum is an identifier for a table in the sysbench schema.
+type tableNum uint64
+
+// rowID is an identifier for a row in a sysbench table.
+type rowID uint64
+
+// kValue is the value of the k column in a sysbench row.
+type kValue uint64
+
+// cValue is the value of the c column in a sysbench row.
+type cValue string
+
+// padValue is the value of the pad column in a sysbench row.
+type padValue string
+
+func randTableNum(rng *rand.Rand) tableNum {
+	return tableNum(rng.Intn(sysbenchTables))
+}
+
+func randRowID(rng *rand.Rand) rowID {
+	return rowID(rng.Intn(sysbenchRowsPerTable))
+}
+
+func randKValue(rng *rand.Rand) kValue {
+	return kValue(rng.Intn(sysbenchRowsPerTable))
+}
+
+func randCValue(rng *rand.Rand) cValue {
+	return cValue(randutil.RandString(rng, 119, randutil.PrintableKeyAlphabet))
+}
+
+func randPadValue(rng *rand.Rand) padValue {
+	return padValue(randutil.RandString(rng, 59, randutil.PrintableKeyAlphabet))
+}
+
+// sysbenchDriver is capable of running sysbench.
+type sysbenchDriver interface {
+	// Transaction orchestration operations.
+	Begin()
+	Commit()
+	// Read-only operations.
+	PointSelect(tableNum, rowID)
+	SimpleRange(tableNum, rowID, rowID)
+	SumRange(tableNum, rowID, rowID)
+	OrderRange(tableNum, rowID, rowID)
+	DistinctRange(tableNum, rowID, rowID)
+	// Read-write operations.
+	IndexUpdate(tableNum, rowID)
+	NonIndexUpdate(tableNum, rowID, cValue)
+	DeleteInsert(tableNum, rowID, kValue, cValue, padValue)
+
+	// Prepares the schema and connection for the workload.
+	prep(rng *rand.Rand)
+}
+
+const (
```

Comment:
Put a permalink to the corresponding stmt in the sysbench source in an inline comment behind each of these queries?

Reply:
A comment that points readers at the `sysbenchOltp*` methods for how these are arranged in sysbench workloads could be helpful.